packages/office-ui-fabric-react/src/utilities/router/Router.tsx

```diff
@@ -93,11 +93,17 @@ function _match(currentPath: string, child: any): boolean {
     path = path || '';
     currentPath = currentPath || '';
 
+    // Use regex to search for a query in the currentPath, then slice out the query to find the correct page to render
+    let regex = new RegExp(path + '(\\?.*)'),
+      result = regex.exec(currentPath),
+      questionMarkIndex = result ? result[0].indexOf('?') : 0,
+      slicedPath = result ? result[0].slice(0, questionMarkIndex) : '';
```

Comment:
Awesome! Minor note - could the regex step be skipped? Since essentially the query needs to be stripped out, maybe go straight to checking for `indexOf('?')`. If > -1, then call `slice`, otherwise leave `currentPath` as is. 

Also, might make sense to move the slicing to `_getPath()` so that the path isn't sliced on every loop (of line 45).
 Probably nil effect on performance but still a minor optimization.

Reply:
Great thoughts! I will poke around at refactoring based on your suggestions.  Thanks!